packages/hash/frontend/src/pages/[account-slug]/types/entity-type/[entity-type-id].page/definition-tab/link-list-card.tsx

```diff
@@ -28,15 +35,61 @@ import {
 } from "./shared/entity-type-table";
 import { InsertTypeRow, InsertTypeRowProps } from "./shared/insert-type-row";
 import { QuestionIcon } from "./shared/question-icon";
+import {
+  generateInitialTypeUri,
+  TypeForm,
+  TypeFormDefaults,
+  TypeFormModal,
+  TypeFormProps,
+  useGenerateTypeBaseUri,
+} from "./shared/type-form";
 import { TYPE_MENU_CELL_WIDTH, TypeMenuCell } from "./shared/type-menu-cell";
 import { useStateCallback } from "./shared/use-state-callback";
 
+const formDataToEntityType = (data: TypeFormDefaults) => ({
+  type: "object" as const,
+  kind: "entityType" as const,
+  title: data.name,
+  description: data.description,
+  allOf: [
+    {
+      $ref: linkEntityTypeUri,
+    },
+  ],
+  properties: {},
+});
+
+export const LinkTypeForm = (props: TypeFormProps) => {
+  const { getEntityType } = useBlockProtocolGetEntityType();
+  const generateTypeBaseUri = useGenerateTypeBaseUri("entity-type");
+
+  const nameExists = async (name: string) => {
+    const entityTypeId = generateInitialTypeUri(generateTypeBaseUri(name));
+
+    const res = await getEntityType({
+      data: {
+        entityTypeId,
+        graphResolveDepths: {
+          constrainsValuesOn: { outgoing: 0 },
+          constrainsPropertiesOn: { outgoing: 0 },
+        },
+      },
+    });
+
+    return !res.data || !!getEntityTypeById(res.data, entityTypeId);
```

Comment:
Added a `@todo` on this line with the question

Reply:
This is copied from the name exists validation for properties – and this has been there ever since that was first introduced in https://github.com/hashintel/hash/commit/543b142dddc3d762e87c240786748fabf549faa9 (this commit was squashed merged in https://github.com/hashintel/hash/commit/fd4520446612b36c77345a4884636fa4cbed142a
)

This doesn't actually answer your question about why I did that way back then.

`data` should never not exist – because this returns a subgraph it should just return an empty one for an entity type id which doesn't exist. In the event `data` isn't there, it'll be some system error I believe (though I'm just guessing at this point). I think this was not adding proper error handling for that eventuality. 

We should definitely come back and fix that, and I'll update the todo (and put it in both places) with this detail.